libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/OAuth/CSFSalesforceOAuthRefresh.m

```diff
@@ -97,6 +97,12 @@ - (void)oauthCoordinatorDidAuthenticate:(SFOAuthCoordinator *)coordinator authIn
     self.network.account.credentials = coordinator.credentials;
     CSFOAuthTokenRefreshOutput *output = [[CSFOAuthTokenRefreshOutput alloc] initWithCoordinator:coordinator];
     [self finishWithOutput:output error:nil];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter] postNotificationName:kSFAuthenticationManagerFinishedNotification
```

Comment:
This came from @jhneves or @ptiarks originally. The notification needs to be there for sure, if the debate is about the thread. Are there concerns about scheduling it on the main thread?

I believe this rule of thumb became popular as the thread that a registered selector executes on is determined by the thread which calls postNotification, this can be an issue for selectors that use UIKit. If this should be the posters responsibility or the recipients is up for discussion. I feel it should be the recipients as only they know what action they will take in response to the notification.